riscv_isac/coverage.py

```diff
@@ -977,6 +999,64 @@ def write_fn_csr_comb_covpt(csr_reg):
                 else:
                     return int(csr_regfile[csr_reg],16)
 
+            def check_label_address(label):
+                return utils.collect_label_address(elf, label)
+
+            def get_mem_val(addr, bytes):
+                if addr in mem_vals:
+                    return mem_vals.get(addr)
+                else:
+                    mem_val = utils.get_value_at_location(elf_path = elf, location = addr, bytes = bytes)
+                    if mem_val is not None:
+                        mem_vals[addr] = mem_val
+                        return mem_val
+                    else:
+                        return None
+
+            def get_pte(pa, pte_addr, pgtb_addr):
+                if instr_vars['xlen'] == 64:
+                    pte_size = 8
+                elif instr_vars['xlen'] == 32:
+                    pte_size = 4
+                if (pgtb_addr >> 12) == (pte_addr >> 12):
+                    if ((pa >> 12) == (get_mem_val(pte_addr, pte_size) >> 10)):
+                        return get_mem_val(pte_addr, pte_size)
+                    else:
+                        return None
+                else:
+                    return None
+
+            def get_pte_prop(prop_name,pa, pte_addr, pgtb_addr):
+                pte_per = get_pte(pa, pte_addr, pgtb_addr)
+                if pte_per is not None:
+                    pte_per = get_pte(pa, pte_addr, pgtb_addr) & 0x3FF
+                    prop_name_lower = prop_name.lower()
+                    if prop_name_lower == 'v' and (pte_per & 0x01 != 0):
+                        return 1
```

Comment:
Don't we have constant definition we can use for all the different bit positions, e.g. #define pte_x_msk 0x08

Reply:
In general, hex (or decimal) constants are bad, (zero is excluded from that broad statement.
You could also define the bit(variable, pos, length) function which extracts a field from a variable and right justifies it.